[Docs] Move snapshot/restore reference content to elasticsearch reference (#144481)
elastic/elasticsearch: docs/reference/elasticsearch/configuration-reference/fs-repository-settings.md

---
navigation_title: Shared file system repository
applies_to:
  deployment:

>     self:

Reviewer: ```suggestion
    self:
    eck:
```

ECK is also considered self-managed hardware and could easily integrate this type of repository.

This one is a good example of a repo that is completely unavailable in ECE or ECH, as the users cannot really tweak the filesystems presented to Elasticsearch nodes.

---
navigation_title: Shared file system repository
applies_to:
  deployment:
    self:
    eck:
---

# Shared file system repository settings [repository-fs-settings]

You can use a shared file system as a repository for [Snapshot and restore](docs-content://deploy-manage/tools/snapshot-and-restore.md). You register repositories via the [Create or update a snapshot repository](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-snapshot-create-repository) API.

There are two categories of settings:

> - [{{es}} settings](#repository-fs-node-settings) control which filesystem paths nodes may use for `fs` repositories. You configure them in [`elasticsearch.yml`](docs-content://deploy-manage/stack-settings.md) on each node (they are not [repository settings](#repository-fs-repository-settings) passed in the snapshot repository API).

Reviewer: ```suggestion
- [{{es}} settings](#repository-fs-node-settings) control which filesystem paths `fs` repositories can use. You configure them in [`elasticsearch.yml`](docs-content://deploy-manage/stack-settings.md) on each node.
```

- [Repository settings](#repository-fs-repository-settings) control each registered repository. You pass them in the `settings` object when you create or update a repository.

## {{es}} settings [repository-fs-node-settings]

$$$path-repo$$$

`path.repo`

> :   ([Static](docs-content://deploy-manage/stack-settings.md#static-cluster-setting)) List of filesystem paths that {{es}} may use for `fs` snapshot repositories. Each repository `location` must resolve to a path under one of these entries on every master and data node. Defaults to an empty list. To register a shared filesystem repository, mount the storage at the same location on each node, add that path (or a parent directory) to `path.repo`, and apply the change according to your deployment. For step-by-step setup, mount paths, and platform-specific examples, refer to [Shared file system repository](docs-content://deploy-manage/tools/snapshot-and-restore/shared-file-system-repository.md).

Reviewer: ```suggestion
:   ([Static](docs-content://deploy-manage/stack-settings.md#static-cluster-setting)) List of filesystem paths that {{es}} can use for `fs` snapshot repositories. Each repository `location` must resolve to a path under one of these entries on every master and data node. Defaults to an empty list.

    To register a shared filesystem repository, mount the storage at the same location on each node and add that path (or a parent directory) to `path.repo`. For step-by-step setup, mount paths, and platform-specific examples, refer to [Shared file system repository](docs-content://deploy-manage/tools/snapshot-and-restore/shared-file-system-repository.md).
```
